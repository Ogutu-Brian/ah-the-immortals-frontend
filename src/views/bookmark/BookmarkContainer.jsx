import React, {Component} from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import BookmarkComponent from '../../components/bookmark/BookmarkComponent';
import BookmarkAction from '../../redux/actions/Bookmark.actions';
import UnBookmarkAction from '../../redux/actions/UnBookmark.actions';

class BookmarkContainer extends Component {
  static  propTypes = {
    article: PropTypes.object.isRequired,
    bookmarkAction: PropTypes.object.isRequired,
    unbookmarkAction: PropTypes.object.isRequired,

  };
  onClick = async (slug) => {
    const { article, bookmarkAction, unbookmarkAction }=this.props;
    if(article.bookmarked)
      await unbookmarkAction(slug);
    else {
      await bookmarkAction(slug);
    }
  };
  render() {
    const { article } = this.props;
    return (
      <div>
        <BookmarkComponent
          article={article}
          handleClick={this.onClick}
        />
      </div>
    );
  }
}

BookmarkContainer.propTypes = {};
const mapStateToProps=({bookmark})=>{
  return{
    bookmark
  };
};
export default connect(mapStateToProps, {bookmarkAction:BookmarkAction, unbookmarkAction:UnBookmarkAction})(BookmarkContainer);
