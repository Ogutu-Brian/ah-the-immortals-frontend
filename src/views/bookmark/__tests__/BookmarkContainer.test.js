import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import BookmarkContainer from '../BookmarkContainer';


describe('Test BookmarkContainer', () => {
  let testStore = configureMockStore([thunk]);
  let store = testStore({});
  const props = {
    article: {
      bookmarked: true
    },
    handleClick : jest.fn(),
  };
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <BookmarkContainer {...props} />
      </BrowserRouter>
    </Provider>
  );

  it('test view profile container snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('test simulates click', () => {
    const clickbookmark = wrapper.find('BookmarkComponent');
    const click = wrapper.find('.fas');
    click.simulate('click');
    expect(props.handleClick.mock.calls.length).toEqual(0);
  });
});
