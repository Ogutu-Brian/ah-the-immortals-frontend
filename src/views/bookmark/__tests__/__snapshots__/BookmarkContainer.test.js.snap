// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BookmarkContainer test view profile container snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(BookmarkContainer)
        article={
          Object {
            "bookmarked": true,
          }
        }
        handleClick={[MockFunction]}
      >
        <BookmarkContainer
          article={
            Object {
              "bookmarked": true,
            }
          }
          bookmarkAction={[Function]}
          handleClick={[MockFunction]}
          unbookmarkAction={[Function]}
        >
          <div>
            <BookmarkComponent
              article={
                Object {
                  "bookmarked": true,
                }
              }
              handleClick={[Function]}
            >
              <div
                className="bookmarked-icon"
              >
                <i
                  className="fas fa-bookmark"
                  onClick={[Function]}
                />
              </div>
            </BookmarkComponent>
          </div>
        </BookmarkContainer>
      </Connect(BookmarkContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
